### PR TITLE
Copy and strip dSYM files for frameworks

### PR DESCRIPTION
**Pull request by eliperkins**
iTunes Connect submissions will validate dSYMs and will warn for dSYMs that contains slices for architectures that are not contained in the app submission.

carthage copy-frameworks now copies dSYM files from the Carthage/Build directory for each framework.

Closes #1021

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -176,17 +175,17 @@ It is possible to use travis-ci in order to build and upload your tagged release
 1. Create `.travis.yml` file at the root of your repository based on that template. Set `FRAMEWORK_NAME` to the correct value.
 
 	Replace PROJECT_PLACEHOLDER and SCHEME_PLACEHOLDER
-
```
Comment (eliperkins) [inline README.md:179]: Remove whitespace

File: Source/CarthageKit/Xcode.swift [hunk 2]
```diff
@@ -1543,14 +1568,23 @@ private func UUIDsFromDwarfdump(URL: NSURL) -> SignalProducer<Set<NSUUID>, Carth
...
-	let bundle = NSBundle(path: frameworkURL.path!)
+private func binaryURL(bundleURL: NSURL) -> Result<NSURL, CarthageError> {
+	let bundle = NSBundle(path: bundleURL.path!)
 
-	if let binaryName = bundle?.objectForInfoDictionaryKey("CFBundleExecutable") as? String {
-		return .Success(frameworkURL.URLByAppendingPathComponent(binaryName))
-	} else {
-		return .Failure(.ReadFailed(frameworkURL, nil))
+	switch bundle?.objectForInfoDictionaryKey("CFBundlePackageType") as? String {
+	case "FMWK"?:
```
Comment (eliperkins) [inline Source/CarthageKit/Xcode.swift]: Make this into an enum

File: Source/CarthageKit/Xcode.swift [hunk 3]
```diff
@@ -1395,59 +1402,77 @@ public func architecturesInFramework(frameworkURL: NSURL) -> SignalProducer<Stri
...
-							.filter { !$0.isEmpty }
-
-						if let components = components {
-							return SignalProducer(values: components)
-						}
-					}
+/// Returns a signal of all architectures present in a given dSYM.
+public func architecturesInDSYM(dSYMURL: NSURL) -> SignalProducer<String, CarthageError> {
+	if let DWARFName = dSYMURL.URLByDeletingPathExtension?.URLByDeletingPathExtension?.lastPathComponent {
+		let DWARFURL = dSYMURL.URLByAppendingPathComponent("Contents/Resources/DWARF/\(DWARFName)")
```
Comment (hyperspacemark) [inline Source/CarthageKit/Xcode.swift]: Let's rename `DWARFName` to `binaryName` and `DWARFURL` to `binaryURL` for consistency with `architecturesInFramework()`.

File: Source/CarthageKit/Xcode.swift [hunk 4]
```diff
@@ -1395,59 +1402,77 @@ public func architecturesInFramework(frameworkURL: NSURL) -> SignalProducer<Stri
...
+		return SignalProducer(value: DWARFURL)
```
Comment (eliperkins) [inline Source/CarthageKit/Xcode.swift]: This seems a bit overkill

File: Source/carthage/CopyFrameworks.swift [hunk 5]
```diff
@@ -27,23 +27,43 @@ public struct CopyFrameworksCommand: CommandType {
...
+						let copydSYMs = copyDebugSymbolsForFramework(source, validArchitectures: validArchitectures)
+						return combineLatest(copyFrameworks, copydSYMs)
+							.then(.empty)
 					}
 			}
 			.waitOnCommand()
 	}
 }
 
+private func copyFrameworkFromURL(source: NSURL, target: NSURL, validArchitectures: [String]) -> SignalProducer<(), CarthageError> {
```
Comment (hyperspacemark) [inline Source/carthage/CopyFrameworks.swift]: Let's rename this to `copyFramework` so it matches the other methods.

File: Source/CarthageKit/Xcode.swift (hunk 4 above)
Comment (eliperkins) [inline Source/CarthageKit/Xcode.swift]: We just return the signal of `architecturesInBinary` of it anyway

**Pull request closed without merging**